apps/mobile/src/screens/GalleryScreen/GalleryEditorNftSelector.tsx

```diff
@@ -29,20 +35,98 @@ export function GalleryEditorNftSelector() {
 
   const route = useRoute<RouteProp<RootStackNavigatorParamList, 'NftSelectorGalleryEditor'>>();
 
+  const [isMultiselectMode, setIsMultiselectMode] = useState(false);
+
+  const [selectedTokens, setSelectedTokens] = useState<SelectedItemMultiMode[]>([]);
+
   const handleSelectNft = useCallback(
     (tokenId: string) => {
-      navigation.navigate({
-        name: 'GalleryEditor',
-        params: {
-          galleryId: route.params.galleryId,
-          stagedTokens: [tokenId],
-        },
-        merge: true,
-      });
+      if (isMultiselectMode) {
+        setSelectedTokens((prevTokens) => {
+          if (
+            prevTokens.some((token) => {
+              return token.id === tokenId;
+            })
+          ) {
+            return [
+              ...prevTokens.filter((token) => {
+                return token.id !== tokenId;
+              }),
+            ];
+          } else {
+            return [
+              ...prevTokens,
+              {
+                id: tokenId,
+              },
+            ];
+          }
+        });
```

Comment:
depending on the number of tokens, might be faster to store `selectedTokens` as a `Set` instead of an array. will need to convert both ways though

Reply:
That smart! Its getting messy with the array. I've replaced it to used `Set`